packages/core/src/baseclient.ts

```diff
@@ -791,11 +794,40 @@ function _validateBeforeSendResult(
         return event;
       },
       e => {
-        throw new SentryError(`\`${beforeSendProcessorName}\` rejected with ${e}`);
+        throw new SentryError(`${beforeSendLabel} rejected with ${e}`);
       },
     );
   } else if (!isPlainObject(beforeSendResult) && beforeSendResult !== null) {
     throw new SentryError(invalidValueError);
   }
   return beforeSendResult;
 }
+
+/**
+ * Process the matching `beforeSendXXX` callback.
+ */
+function processBeforeSend<T extends Event>(
+  options: ClientOptions,
+  event: T,
+  hint: EventHint,
+): PromiseLike<T | null> | T | null {
+  const { beforeSend, beforeSendTransaction } = options;
+
+  if (isErrorEvent(event) && beforeSend) {
+    return beforeSend(event, hint) as PromiseLike<T | null> | T | null;
```

Comment:
As discussed: let's remove this type cast and the generic in this function because it might actually lead to errors